src/plugins/irc-events/link.js

```diff
@@ -313,71 +310,89 @@ function getRequestHeaders(headers) {
 	return formattedHeaders;
 }
 
-function fetch(uri, headers, cb) {
-	let req;
+function fetch(uri, headers) {
+	// Stringify the object otherwise the objects won't compute to the same value
+	const cacheKey = JSON.stringify({uri, headers});
```

Comment:
Probably should be an array instead of an object to reduce length of the string.

Reply:
That would go from:
`'{"uri":"https://example.com","headers":{"foo":"bar"}}'`
to:
`'["https://example.com",{"foo":"bar"}]'`
Correct?

If so, that saves us 16 characters per string, I wouldn't worry too much.
I'm obviously not opposed if you prefer to do that though.

Reply:
Worth it!